doc/users/interactive.rst

```diff
@@ -1,12 +1,299 @@
+
+
+.. _key-event-handling:
+
+Navigation Keyboard Shortcuts
+-----------------------------
+
+The following table holds all the default keys, which can be
+overwritten by use of your :ref:`matplotlibrc
+<sphx_glr_tutorials_introductory_customizing.py>`.
+
+================================== ===============================
+Command                            Default key binding and rcParam
+================================== ===============================
+Home/Reset                         :rc:`keymap.home`
+Back                               :rc:`keymap.back`
+Forward                            :rc:`keymap.forward`
+Pan/Zoom                           :rc:`keymap.pan`
+Zoom-to-rect                       :rc:`keymap.zoom`
+Save                               :rc:`keymap.save`
+Toggle fullscreen                  :rc:`keymap.fullscreen`
+Toggle major grids                 :rc:`keymap.grid`
+Toggle minor grids                 :rc:`keymap.grid_minor`
+Toggle x axis scale (log/linear)   :rc:`keymap.xscale`
+Toggle y axis scale (log/linear)   :rc:`keymap.yscale`
+Close Figure                       :rc:`keymap.quit`
+Constrain pan/zoom to x axis       hold **x** when panning/zooming with mouse
+Constrain pan/zoom to y axis       hold **y** when panning/zooming with mouse
+Preserve aspect ratio              hold **CONTROL** when panning/zooming with mouse
+================================== ===============================
+
+
+.. _other-shells:
+
+Other Python prompts
+====================
+
+If you can not or do not want to use IPython, interactive mode works
+in the vanilla python prompt
+
+
+.. sourcecode:: pycon
+
+   >>> import matplotlib.pyplot as plt
+   >>> plt.ion()
+   >>>
+
+however this does not ensure that the event hook is properly installed
+and your figures may not be responsive.  Please consult the
+documentation of your GUI toolkit for details.
+
+
+
+Jupyter Notebooks / Lab
+-----------------------
+
+.. note::
+
+   To get the interactive functionality described here, you must be
+   using an interactive backend.  The default backend in notebooks,
+   the inline backend, is not.  `~ipykernel.pylab.backend_inline`
+   renders the figure once and inserts a static image into the
+   notebook when the cell is executed.  The images are static and can
+   not be panned / zoomed, take user input, or be updated from other
+   cells.
+
+To get interactive figures in the 'classic' notebook or jupyter lab
+use the `ipympl <https://github.com/matplotlib/ipympl>`__ backend
+(must be installed separately) which uses the **ipywidget** framework.
+If ``ipympl`` is installed use the magic:
+
+.. sourcecode:: ipython
+
+  %matplotlib widget
+
+to select and enable it.
+
+If you only need to use the classic notebook you can use
+
+.. sourcecode:: ipython
+
+  %matplotlib notebook
+
+which uses the `.backend_nbagg` backend which ships with Matplotlib.
+However nbagg does not work in Jupyter Lab due to changes in the front
```

Comment:
Maybe delete "due to changes...".  It's not clear what is meant by "the front end".